### ISSUE TITLE
Using Boost in clSPARSE

### ISSUE THREAD

**Issue by kknox** [clMathLibraries/clSPARSE#23]
Currently, we use the boost library in our benchmark and our testing applications.  I've seen a few places now in our library where boost would be nice to have: for instance the filesystem library.  The filesystem library in the compilers implementation are not standard yet, some use filesystem v2 and others use filesystem v3.  Using boost would remove the compiler dependency on the implementation.  

How would users feel about requiring boost to build the library proper?  I feel like Boost is an 'pseudo' c++ standard that the only negative is that it doesn't ship with the compiler. Cmake removes almost all of the pain of building boost, but I've run into users that just hate external dependencies on principle.

**Comment by jpola** [comment]
@kknox, Your version of mtx readers are very good at the moment. We tested them with plenty of matrices. We support the types of matrices we want. Do we still want to include boost into the internal part of the clSPARSE?

Im very happy with the way how the boost is included in the build, test and benchmarks infrastructure. I think that we shouldn't include boost into internal code. If you agree with me I think we can close this discussion. :smile:

**Comment by kknox** [comment]
@jpola 
I've never been able to make a final decision; it would be hard to remove boost from the library if we decided to start using it.  However, the reverse is not true; it would be easy to use boost at any time once we had a real need for it.  

My big hesitation is that we did not have a BIG need for it, just a few lines in the mtx reader would benefit to verify file extensions.  I think I am good to close this issue, with the option to re-evaluate in the future if there is some killer feature in boost that would make the library easier to develop/maintain.
